web/src/dialog_widget.ts

```diff
@@ -152,6 +155,34 @@ export function launch(conf: WidgetConfig): void {
 
     const $submit_button = $dialog.find(".dialog_submit_button");
 
+    if (conf.update_submit_on_state_change) {
+        const original_values: {[key: string]: unknown} = {};
+        const current_values: {[key: string]: unknown} = {};
+
+        $(() => {
```

Comment:
Is it necessary to include the code below inside `$()`? The click handler is defined normally below though.

Reply:
it checks that if the document is ready then it attaches click handlers to rendered input fields. not including this was causing errors sometimes as the code was trying to attach the click handlers to elements that were not present in DOM.

I have checked the updated code just now by removing this it is not giving an error but should we keep this to ensure that DOM is ready?

Reply:
@sahil839 I will also update the commit message and push the code as soon as you confirm this.

Reply:
Ok, let's keep it as it is for now and can update if required after feedback from others.

Reply:
As I mentioned above:
> I have checked the updated code just now by removing this it is not giving an error anymore

Hence, I removed `$()` in the updated PR after testing it several times. It is working properly every time. Do share your feedback in case you find the issue in your manual testing.